### ISSUE TITLE
[Content] Incorrect cross_entropy_loss function

### ISSUE THREAD

**Issue by ry2sk** [Qiskit/platypus#1341]
### Where does the issue happen?

https://github.com/Qiskit/platypus/blob/main/notebooks/quantum-machine-learning/vqc.ipynb

### What is the content issue?

In the cross_entropy_loss function, considering the relationship between `expected` and `p`, 
`return -(expected*np.log(p)+(1-expected)*np.log(1-p))` should be just `return -np.log(p)`.

### Additional information

This problem becomes apparent when the number of train and test data is increased to 50 and 20 respectively.

**Comment by frankharkins** [comment]
@ry2sk we discussed with a content matter expert and we believe `cross_entropy_loss` is correct - can you elaborate on:
> This problem becomes apparent when the number of train and test data is increased to 50 and 20 respectively.

please?

**Comment by ry2sk** [comment]
1. Change training_size and test_size to 50 and 20 respectively.
`TRAIN_DATA, TRAIN_LABELS, TEST_DATA, TEST_LABELS = (`
`    ad_hoc_data(training_size=50,`
`                test_size=20,`
2.  Fix the following code to avoid underflow.
`return -(expected*np.log(p)+(1-expected)*np.log(1-p))`
--> `return -(expected*np.log(p + 1e-16)+(1-expected)*np.log(1-p + 1e-16))`
3. Run the notebook then you can get test accuracy 0.5.
4. Change the previous code as follows.
`return -np.log(p)`
5. Run the notebook again then you can get test accuracy 1.0.

**Comment by ry2sk** [comment]
@frankharkins
 If you want to keep the binary cross entropy style, the following code is correct.
`p = classification.get(1)  # Prob. of correct classification`
`return -(expected*np.log(p)+(1-expected)*np.log(1-p))`

**Comment by frankharkins** [comment]
@ry2sk thanks for your reply. I'm still a bit confused, other than adding `1e-16` for the underflow, I can't see the difference between your most recent comment and the original code (show below):
```
p = classification.get(expected)  # Prob. of correct classification
return -(expected*np.log(p)+(1-expected)*np.log(1-p))
```
Would fixing the underflow resolve this issue for you?

**Comment by ry2sk** [comment]
The difference between the modified version and the original code is that the classification.get parameter is 1.

**Comment by frankharkins** [comment]
Hi @ry2sk, thanks for bearing with us, we see the problem now and I've made a PR #1417 to fix it.
